core/activity/src/provider/service.rs

```diff
@@ -74,7 +77,60 @@ pub fn bind_gsb(db: &DbExecutor, tracker: TrackerRef) {
     counter!("activity.provider.destroyed.by_requestor", 0);
     counter!("activity.provider.destroyed.unresponsive", 0);
 
-    local::bind_gsb(db, tracker);
+    local::bind_gsb(&db.clone(), tracker.clone());
+
+    tokio::task::spawn_local(initial_checkup(db.clone(), tracker.clone()));
+}
+
+async fn initial_checkup(db: DbExecutor, tracker: TrackerRef) {
+    log::info!("Initial activities checkup");
+    let agreements_res = get_agreements_by_state(AgreementState::Approved).await;
+    let Ok(agreements) = agreements_res else {
+        log::error!("Acitivties checkup error. Failed to list Agreements. Err: {agreements_res:?}");
+        return;
+    };
+
+    for agreement in agreements {
+        let agreement_res = bus::service(market::BUS_ID)
+            .send(market::GetAgreement::as_provider(agreement.id))
+            .await;
+        let Ok(Ok(agreement)) = agreement_res else {
+            log::error!(
+                "Acitivties checkup error. Failed to get Agreement. Err: {agreement_res:?}"
+            );
+            continue;
+        };
+        if let Err(err) = monitor_alive_activities(agreement, db.clone(), tracker.clone()).await {
+            log::error!("Acitivties initial checkup error. Err: {err}")
+        };
+    }
+}
+
+async fn monitor_alive_activities(
+    agreement: Agreement,
+    db: DbExecutor,
+    tracker: TrackerRef,
+) -> anyhow::Result<()> {
+    log::info!(
+        "Initial activities checkup for agreement {}",
+        agreement.agreement_id
+    );
+    let activities_ids = get_activities_for_agreement(&db, &agreement.agreement_id).await?;
+    let activity_state_dao = db.as_dao::<ActivityStateDao>();
+    for activity_id in activities_ids {
+        let activity_state = activity_state_dao.get(&activity_id).await?;
+        if is_responsive(activity_state) {
+            log::info!("Initial activty {activity_id} monitoring");
+            tokio::task::spawn_local(monitor_activity(
+                db.clone(),
+                tracker.clone(),
+                activity_id,
+                *agreement.provider_id(),
+                None,
+            ));
+        }
```

Comment:
Initial spawn of `monitor_activity` changes not `Terminated` activities into `Unresponsive` activities, so it makes sense to spawn it only for activities which are not `Terminated` and not `Unresponsive` (`ya-client` provides `alive()` function, but for this usecase only I created more specific `is_responsive()` function. I did not wanted to update `ya-client` only to create this function)